cmd/kubebuilder/create/controller/run.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	createutil "github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder/create/util"
+	generatecmd "github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder/generate"
+	"github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder/util"
+	"github.com/markbates/inflect"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+var nonNamespacedKind bool
+var generate bool
+var CoreType bool
+
```

Comment:
We should move away from this pattern of defining pkg level globals for command options. Currently all commands are implemented like this but its an anti-pattern where push down cobra.command and these globals too deep in to the business logic.

Idea is to define a separate struct for command args:
`
type controllerArgs struct {
 namespaced bool
 coreType bool
 generate bool
}
`
Bind above struct field to the command-args. So filling of struct is delegated to cobra with some basic validation.

Then have validate() method on controllerArgs

And then execution should NOT take cobra.cmd or args as input like we do today. 

Not a deal breaker, but wanted to bring it out (can be handled in separate PR but will be good if we can do it in current PR because then we can point people to the pattern here to refactor commands):

Reply:
I can do it in this PR.

Reply:
Done